app/src/main/java/com/xmartlabs/daydreaming/ui/LoginRegisterFragment.java

```diff
@@ -86,17 +97,38 @@ void onClickedLogin() {
 
   @OnClick(R.id.create_login_textview)
   void onClickedCreateLogin() {
+    setupErrorFieldsAsInvisible();
     //TODO create user and login user
     if (isRegisterSelected) {
-      if (passwordFieldView.getText().toString().length() < 5) {
+      checkFieldsDuringRegistration();
+    } else {
+      checkFieldsDuringLogin();
+    }
+  }
+
+  private void checkFieldsDuringLogin() {
+    if (!(usernameFieldView.getText().toString().toLowerCase()).equals(getString(USER))) {
```

Comment:
you could put this in a function as: `isValidUsername()`. This will be cleaner, plus it would help later on if this is a TODO to check the username with db or service or something like that. Same case with password validation